### PR TITLE
Add container basic test case

### PR DESCRIPTION
**Pull request by xlgao-zju**
cc @feiskyer @Random-Liu 

Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

### CODE REVIEW

File: pkg/framework/util.go [hunk 1]
```diff
@@ -31,10 +31,13 @@ import (
...
-	defaultUIDPrefix       string = "e2e-cri-uid"
-	defaultNamespacePrefix string = "e2e-cri-namespace"
-	defaultAttempt         uint32 = 2
+	defaultAPIVersion           string = "0.1.0"
+	defaultUIDPrefix            string = "e2e-cri-uid"
+	defaultNamespacePrefix      string = "e2e-cri-namespace"
+	defaultAttempt              uint32 = 2
+	defaultContainerImage       string = "gcr.io/google_containers/busybox:1.24"
+	defaultStopContainerTimeout int64  = 60
+	defaultExecSyncTimeout      string = "5s"
```
Comment (feiskyer) [inline pkg/framework/util.go]: Why not make `defaultExecSyncTimeout` as `int64` (same with `defaultStopContainerTimeout `)?

File: pkg/framework/util.go [hunk 2]
```diff
@@ -218,9 +221,151 @@ func ListPodSanboxForID(c internalapi.RuntimeService, podID string) []*runtimeap
...
+	By("List PodSandbox.")
 	pods, err := c.ListPodSandbox(filter)
 	ExpectNoError(err, "failed to list PodSandbox status: %v", err)
 	Logf("List PodSandbox succeed")
 	return pods
 }
+
+// ContainerFound returns whether containers is found.
+func ContainerFound(containers []*runtimeapi.Container, containerID string) bool {
+	if len(containers) == 1 && containers[0].Id == containerID {
```
Comment (feiskyer) [inline pkg/framework/util.go]: what if `len(containers)>1` ?

File: pkg/validate/container.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+var _ = framework.KubeDescribe("Container", func() {
+	f := framework.NewDefaultCRIFramework()
+
+	var c internalapi.RuntimeService
+
+	BeforeEach(func() {
+		c = f.CRIClient.CRIRuntimeClient
+	})
+	Context("runtime should support basic operations on cotainer", func() {
```
Comment (feiskyer) [inline pkg/validate/container.go]: s/cotainer/container

**Comment by xlgao-zju** [comment]
@feiskyer Updated

File: pkg/framework/util.go [hunk 4]
```diff
@@ -31,9 +31,12 @@ import (
 )
 
 const (
-	defaultUIDPrefix       string = "e2e-cri-uid"
-	defaultNamespacePrefix string = "e2e-cri-namespace"
-	defaultAttempt         uint32 = 2
+	defaultUIDPrefix            string = "e2e-cri-uid"
+	defaultNamespacePrefix      string = "e2e-cri-namespace"
+	defaultAttempt              uint32 = 2
+	defaultContainerImage       string = "gcr.io/google_containers/busybox:1.24"
```
Comment (feiskyer) [inline pkg/framework/util.go]: `defaultContainerImage` is not pulled before creating any containers

Comment (xlgao-zju) [inline pkg/framework/util.go]: `dockershim` will pull the image if it does not exist. Is this a `dockershim` specific behavior? And I think runtime should pull the image if it does not exist. It's convenient for users when they create a container which image does not exist. What do you think?

Comment (feiskyer) [inline pkg/framework/util.go]: > I think runtime should pull the image if it does not exist. 

No, this is not consistent with CRI. There is explicit image pull API, which ensures the image is pulled before creating containers. We should not rely on docker's behavior.

Comment (xlgao-zju) [inline pkg/framework/util.go]: OK, will fix this.

**Comment by xlgao-zju** [comment]
@feiskyer Updated

File: pkg/framework/util.go [hunk 5]
```diff
@@ -206,9 +212,205 @@ func ListPodSanboxForID(c internalapi.RuntimeService, podID string) []*runtimeap
...
+	cmd := []string{"echo", "hello"}
+	stdout, stderr, err := c.ExecSync(containerID, cmd, time.Duration(defaultExecSyncTimeout)*time.Second)
+	ExpectNoError(err, "failed to execSync in container %q", containerID)
+	Expect(string(stdout)).To(Equal("hello\n"), "The stdout output of execSync should be %q", stdout)
+	Expect(stderr).To(BeNil(), "The stderr should be nil.")
+	Logf("Execsync succeed")
+}
+
+// PullPublicImage pulls the public image named imageName.
+func PullPublicImage(c internalapi.ImageManagerService, imageName string) {
```
Comment (feiskyer) [inline pkg/framework/util.go]: I think it's better to move those image related functions into `validate/image.go`

Comment (xlgao-zju) [inline pkg/framework/util.go]: Some functions in `framework.go` will need to call these image related functions, such as `CreateContainer`.

**Comment by xlgao-zju** [comment]
@feiskyer Move pod/container/image related functions to their own file.

File: pkg/framework/framework.go [hunk 6]
```diff
@@ -17,10 +17,9 @@ limitations under the License.
 package framework
 
 import (
-	internalapi "k8s.io/kubernetes/pkg/kubelet/api"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	internalapi "k8s.io/kubernetes/pkg/kubelet/api"
```
Comment (feiskyer) [inline pkg/framework/framework.go]: nit: keep this as is.

File: pkg/framework/util.go [hunk 7]
```diff
@@ -21,19 +21,10 @@ import (
...
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	"github.com/pborman/uuid"
-)
-
-const (
-	defaultUIDPrefix       string = "e2e-cri-uid"
-	defaultNamespacePrefix string = "e2e-cri-namespace"
-	defaultAttempt         uint32 = 2
+	"k8s.io/kubernetes/pkg/kubelet/remote"
```
Comment (feiskyer) [inline pkg/framework/util.go]: same as above, separate `.` import and others

File: pkg/validate/pod.go [hunk 8]
```diff
@@ -18,10 +18,16 @@ package validate
 
 import (
 	"github.com/kubernetes-incubator/cri-tools/pkg/framework"
-	internalapi "k8s.io/kubernetes/pkg/kubelet/api"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	internalapi "k8s.io/kubernetes/pkg/kubelet/api"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
```
Comment (feiskyer) [inline pkg/validate/pod.go]: same here

File: pkg/framework/util.go (hunk 7 above)
Comment (xlgao-zju) [inline pkg/framework/util.go]: ACK

**Comment by xlgao-zju** [comment]
> separate . import and others
It's clean. :+1:

**Pull request merged**